tests/all/resin_root_part/resin_root_part_rpi

```diff
@@ -0,0 +1,35 @@
+#!/bin/sh
+#
+
+. /tmp/test-lib.sh
+CONF=/mnt/boot/cmdline.txt
+
+# we expect this to be '2' on a fresh system. '3' is also good.
+current=$(mount | grep 'on / type' | sed -e 's/ .*//' -e 's/^.*\(.\)/\1/') # e.g. 2
+dev=$(mount | grep 'on / type' | sed -e 's/ .*//' -e 's/[0-9]$//') # e.g. /dev/mmcblk1p
+
+# assume $CONF exists and is valid on a rpi
+
+if [ "$current" != "2" ] && [ "$current" != "3" ]; then
+    log ERROR "unexpected root partition number"
+fi
+
+configured=$(sed -e 's/^.* root=[^ ]*\([^ ]\).*/\1/' $CONF) # e.g. 2
+if [ "$current" != "$configured" ]; then
+    rm -f /mnt/boot/.test
+    log FAILED "\$current ($current) != \$configured ($configured)"
+fi
+
+if [ -e /mnt/boot/.test ]; then
+    rm /mnt/boot/.test
+    log PASSED "OK"
+else
+    test "$current" = "2" && other=3 || other=2
+
+    log INFO "copying root partition from $dev$current to $dev$other..."
+    dd "if=$dev$current" "of=$dev$other" bs=1M
+    sed -i -e "s# root=[^ ]*# root=$dev$other#" "$CONF" # $dev$other contains slashes
+    touch /mnt/boot/.test
+    log INFO "rebooting..."
+    /sbin/reboot
+fi
```

Comment:
I considered that, but there isn't that much code duplication here, 10 lines or so. I think it's easier to reason about the code this way. We'll get back to this if there is ever a need to create a third script.